### PR TITLE
Chapter 13: User microposts

### PR DESCRIPTION
**Pull request by nguyenthanhtu95**

### CODE REVIEW

**Comment by nguyenthanhtu95** [comment]
ready

File: app/controllers/microposts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  def destroy
+    @micropost.destroy
+    flash[:success] = t(:micropost_deleted)
+    redirect_to request.referrer || root_url
+  end
+
+  private
+
+    def micropost_params
+      params.require(:micropost).permit(:content, :picture)
```
Comment (phamvanbk11) [inline app/controllers/microposts_controller.rb]: Xóa () sau permit

File: app/views/password_resets/edit.html.erb [hunk 2]
```diff
@@ -4,7 +4,7 @@
 <div class="row">
   <div class="col-md-6 col-md-offset-3">
     <%= form_for(@user, url: password_reset_path(params[:id])) do |f| %>
-      <%= render 'shared/error_messages' %>
+      <%= render 'shared/error_messages', object: f.object  %>
```
Comment (phamvanbk11) [inline app/views/password_resets/edit.html.erb]: nháy "

File: app/views/shared/_micropost_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  </span>
+<% end %>
+
+<script type="text/javascript">
+  $("#micropost_picture").bind("change", function() {
+    var size_in_megabytes = this.files[0].size/1024/1024;
+    if (size_in_megabytes > 5) {
+      alert(t(:maximum_picture_size_alert));
+    }
+  });
```
Comment (phamvanbk11) [inline app/views/shared/_micropost_form.html.erb]: tách ra file js

File: app/views/shared/_micropost_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<%= form_for(@micropost) do |f| %>
+  <%= render "shared/error_messages", object: f.object %>
+  <div class="field">
+    <%= f.text_area :content, placeholder: t(:compose_new_micropost) %>
+  </div>
+  <%= f.submit "Post", class: "btn btn-primary" %>
```
Comment (phamvanbk11) [inline app/views/shared/_micropost_form.html.erb]: I18n nhé

File: app/views/static_pages/home.html.erb [hunk 5]
```diff
@@ -1,14 +1,31 @@
...
+    <aside class="col-md-4">
+      <section class="user_info">
+        <%= render "shared/user_info" %>
+      </section>
+      <section class="micropost_form">
+        <%= render "shared/micropost_form" %>
+      </section>
+    </aside>
+    <div class="col-md-8">
+      <h3>Micropost Feed</h3>
```
Comment (phamvanbk11) [inline app/views/static_pages/home.html.erb]: I18n nhé

File: app/views/users/show.html.erb [hunk 6]
```diff
@@ -8,4 +8,13 @@
       </h1>
     </section>
   </aside>
+  <div class="col-md-8">
+    <% if @user.microposts.any? %>
+      <h3>Microposts (<%= @user.microposts.count %>)</h3>
```
Comment (phamvanbk11) [inline app/views/users/show.html.erb]: I18n nhé

**Comment by phamvanbk11** [comment]
commented

File: app/controllers/microposts_controller.rb (hunk 1 above)
Comment (nguyenthanhtu95) [reply app/controllers/microposts_controller.rb]: fixed

File: app/views/password_resets/edit.html.erb (hunk 2 above)
Comment (nguyenthanhtu95) [reply app/views/password_resets/edit.html.erb]: fixed

File: app/views/shared/_micropost_form.html.erb (hunk 4 above)
Comment (nguyenthanhtu95) [reply app/views/shared/_micropost_form.html.erb]: fixed

File: app/views/shared/_micropost_form.html.erb (hunk 3 above)
Comment (nguyenthanhtu95) [reply app/views/shared/_micropost_form.html.erb]: fixed

File: app/views/static_pages/home.html.erb (hunk 5 above)
Comment (nguyenthanhtu95) [reply app/views/static_pages/home.html.erb]: fixed

File: app/views/users/show.html.erb (hunk 6 above)
Comment (nguyenthanhtu95) [reply app/views/users/show.html.erb]: fixed

**Comment by nguyenthanhtu95** [comment]
ready

**Comment by phamvanbk11** [comment]
merged

**Pull request merged**